Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/RenamingPolicy.cs

```diff
@@ -1,270 +1,249 @@
 ﻿using System.Text.RegularExpressions;
 
-namespace Frends.SFTP.DownloadFiles.Definitions
+namespace Frends.SFTP.DownloadFiles.Definitions;
+
+///<summary>
+/// Policies for creating names for remote files: expands macros etc.
+///</summary>
+internal class RenamingPolicy
 {
-    ///<summary>
-    /// Policies for creating names for remote files: expands macros etc.
-    ///</summary>
-    internal class RenamingPolicy
+    private IDictionary<string, Func<string, string>> MacroHandlers;
+    private IDictionary<string, Func<string, string>> SourceFileNameMacroHandlers;
+
+    public RenamingPolicy() : this("", Guid.Empty)
     {
-        private IDictionary<string, Func<string, string>> MacroHandlers;
-        private IDictionary<string, Func<string, string>> SourceFileNameMacroHandlers;
+    }
 
-        public RenamingPolicy() : this("", Guid.Empty)
+    public RenamingPolicy(string transferName, Guid transferId)
+    {
+        MacroHandlers = InitializeMacroHandlers(transferName, transferId);
+        SourceFileNameMacroHandlers = InitializeSourceFileNameMacroHandlers();
+    }
+
+    public string CreateRemoteFileName(string originalFileName, string remoteFileDefinition)
+    {
+        if (!string.IsNullOrEmpty(remoteFileDefinition) && remoteFileDefinition.Contains("?"))
         {
+            throw new ArgumentException("Character '?' not allowed in remote filename.", "remoteFileDefinition");
         }
 
-        public RenamingPolicy(string transferName, Guid transferId)
+        if (string.IsNullOrEmpty(originalFileName))
         {
-            MacroHandlers = InitializeMacroHandlers(transferName, transferId);
-            SourceFileNameMacroHandlers = InitializeSourceFileNameMacroHandlers();
+            throw new ArgumentException("Original filename must be set.", "originalFileName");
```

Comment:
This can be a one-line if-statement. Also, when you are referring to a variable name in error message, you should use `nameof(originalFileName)`.

Reply:
Fixed.